rosserial_server/include/rosserial_server/utils.h

```diff
@@ -0,0 +1,35 @@
+/******************************************************************************
+ *  Copyright (c) 2024, Intermodalics BVBA                                    *
+ *  All rights reserved.                                                      *
+ ******************************************************************************/
+
+#ifndef ROSSERIAL_SERVER_UTILS_H
+#define ROSSERIAL_SERVER_UTILS_H
+
+#include <filesystem>
+#include <glob.h>
+
+inline std::string getPortNameByPartialPortName(const std::string& port_name){
```

Comment:
nit: rename `getPortNameByPartialPortName()` -> `getPortNameFromPattern()`?

A wildcard "pattern" (the terminology from [glob](https://linux.die.net/man/7/glob)) like `/dev/serial/by-id/usb-Espressif*` is a bit more generic and not just a "partial port name".